gaphor/UML/classes/classconnect.py

```diff
@@ -121,14 +123,24 @@ def connect_subject(self, handle):
                     end2.type is head_type and end1.type is tail_type
                 ):
                     return
+                else:
+                    head_nav = end1.navigability
+                    head_agg = end1.aggregation
+                    tail_nav = end2.navigability
+                    tail_agg = end2.aggregation
 
             # Create new association
             relation = UML.model.create_association(head_type, tail_type)
```

Comment:
Does this go well, or do we end up with stale associations in the model? If the association item already has an association connected it should not create a new association again.

Reply:
@amolenaar I refactored this part to be more clear. I think the UML.Assocation is only created if the line doesn't have a subject already, for example by creating a plain association. What do you think?

Reply:
This is better. There is less confusion on what's done and when. Well done :+1: